receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper.go

```diff
@@ -135,7 +144,7 @@ func (s *scraper) getProcessMetadata() ([]*processMetadata, error) {
 
 		executable, err := getProcessExecutable(handle)
 		if err != nil {
-			errs = append(errs, fmt.Errorf("error reading process name for pid %v: %w", pid, err))
+			errs = append(errs, consumererror.NewPartialScrapeError(fmt.Errorf("error reading process name for pid %v: %w", pid, err), 1))
```

Comment:
Didn't think of this during the previous review but seeing it now a less verbose version might be

```go
var errs consumererror.ScrapeErrors
errs.Add(fmt.Errorf("error reading process name for pid %v: %w", pid, err), 1))
errs.Add(...)
return errs.Combine()
```

Or something. Just an idea.

Reply:
It is a bit ugly yea.

I don't know if your suggestion makes that much difference though. Another idea might be to make a `fmt`-like constructor for `PartialScrapeError` so we can do something like `consumererror.FmtPartialScrapeError(1, "error reading process name for pid %v: %w", pid, err)`

WDYT?

In any case I will probably do in a follow up PR as this one is very large already

Reply:
Makes sense - we would also want a way to include regular unexpected errors that don't necessarily make sense to be represented as a scrape error.

Moved this discussion to an issue: https://github.com/open-telemetry/opentelemetry-collector/issues/2046